File: impl/ruby-4/fizzbuzz.rb
def word(n)
  isfizz = (n % 3).zero?
  isbuzz = (n % 5).zero?
  if isfizz && isbuzz then
    'fizz buzz'
  elsif isfizz then
    'fizz'
  elsif isbuzz then
    'buzz'
  else
    n.to_s
  end
end

def play(n)
  1.upto(n).each { |i| puts word(i) }
end

n = STDIN.read.to_i
play(n)

File: impl/nim-2/fizzbuzz.nim
import std/strutils

proc word(n: int): string =
  let
    isfizz = n mod 3 == 0
    isbuzz = n mod 5 == 0
  if isfizz and isbuzz:
    "fizz buzz"
  elif isfizz:
    "fizz"
  elif isbuzz:
    "buzz"
  else:
    $n

proc say(n: int) =
  echo word n

proc play(n: int) =
  for i in 1..n:
    say i

proc main() =
  play parseInt readAll stdin

main()
